documentation/next-generation-quickstart.md

```diff
@@ -0,0 +1,428 @@
+
+***Import the packages***
+```typescript
+import * as compute from "@azure/arm-compute";
+import * as network from "@azure/arm-network";
+import * as resources from "@azure/arm-resources";
+import { DefaultAzureCredential } from "@azure/identity";
+```
+
+***Define the global variables***
+```typescript
+const subscriptionId = process.env.AZURE_SUBSCRIPTION_ID;
+const resourceGroupName = "testRG";
+const virtualMachineName = "virtualmachinex";
+const subnetName = "subnetnamex";
+const interfaceName = "interfacex";
+const networkName = "networknamex";
+const location = "eastus";
+```
+
+***Authentication and Setup***
+```typescript
+const credential = new DefaultAzureCredential();
+const computeClient = new compute.ComputeManagementClient(credential, subscriptionId);
+const networkClient = new network.NetworkManagementClient(credential, subscriptionId);
+const resourcesClient = new resources.ResourceManagementClient(credential, subscriptionId);
+```
+
+***Creating a Resource Group***
+```typescript
+function createResourceGroup() {
+    const parameter: resources.ResourceGroup = {
+        location: "eastus",
+        tags: {
+            tag1: "value1"
+        }
+    };
+    await resourcesClient.resourceGroups.createOrUpdate(resourceGroupName, parameter).then(
+        result => {
+            console.log(result);
+        }
+    )
+}
+```
+
+***Creating a Virtual Network***
+```typescript
+function createVirtualNetwork() {
+    const parameter: network.VirtualNetwork = {
+        location: location,
+        addressSpace: {
+            addressPrefixes: ['10.0.0.0/16']
+        }
+    };
+    const virtualNetworks_create_info = await networkClient.virtualNetworks.beginCreateOrUpdateAndWait(resourceGroupName, networkName, parameter);
+    console.log(virtualNetworks_create_info);
+}
+```
+
+***Creating a Subnet***
+```typescript
+function createSubnet() {
+    const subnet_parameter: network.Subnet = {
+        addressPrefix: "10.0.0.0/24"
+    };
+    const subnet_create_info = await networkClient.subnets.beginCreateOrUpdateAndWait(resourceGroupName, networkName, subnetName, subnet_parameter);
+    console.log(subnet_create_info)
+}
+```
+
+***Creating a Network Interface***
+```typescript
+function createNetworkInterface(group_name: any, location: any, nic_name: any) {
+    const parameter: network.NetworkInterface = {
+        location: location,
+        ipConfigurations: [
+            {
+                name: "MyIpConfig",
+                subnet: {
+                    id: "/subscriptions/" + subscriptionId + "/resourceGroups/" + resourceGroupName + "/providers/Microsoft.Network/virtualNetworks/" + networkName + "/subnets/" + subnetName
+
+                }
+            }
+        ]
+    };
+    const nic_info = await networkClient.networkInterfaces.beginCreateOrUpdateAndWait(group_name, nic_name, parameter);
+    console.log(nic_info);
+}
+```
+
+***Creating a Virtual Machine***
```

Comment:
Do we need to leave some space for user to figure out how to create vm themselves? Otherwise, they would just copy and paste the source.

Reply:
it is fine for showing this for general users. For user study, we can duplicate this document and remove this section